### DISCUSSION TITLE
Restrict to a single, non-shared child process

### DISCUSSION THREAD

**Discussion by hanzzzzz** [MicrosoftEdge/WebView2Feedback#3904] [Q&A]
Currently, my test application using the WebView2 runtime spawns one child `msedgewebview2.exe`, which again spawns five more 
`msedgewebview2.exe` child processes. Those are shared with new instances of my test application.

In my use case, I need to have one (or more, if required) isolated `msedgewebview2.exe` process which is *not* shared with other processes using the WebView2 runtime. This is required for restricting network traffic to this single process using a firewall, as currently some of the child processes would handle the network traffic, so if I allow it through the firewall, all other processes using the WebView2 runtime would also be able to get through the firewall.

Is it possible to get a dedicated `msedgewebview2.exe` process, which is only used by exactly one instance of my application?

**Comment by champnic** [comment]
Hey @hanzzzzz - We generally match the Edge/Chromium process model:
https://docs.microsoft.com/en-us/microsoft-edge/webview2/concepts/process-model

Browser processes are associated with a user data folder, so if you want separate processes (for example, a "firewall allowed" group and a "non-firewall allowed" group) you could create 2 user data folders, 1 for each group.

How are you planning on allowing the firewall access?
